indexers/social_feed/social_feed_reposts_v9.js

```diff
@@ -0,0 +1,450 @@
+ * context is a global variable that contains helper methods.
+ * context.db is a subfield which contains helper methods to interact with your database.
+ *
+ * Learn more about indexers here:  https://docs.near.org/concepts/advanced/indexers
+ *
+ * @param {block} Block - A Near Protocol Block
+ */
+async function getBlock(block: Block) {
+  function base64decode(encodedValue) {
+    let buff = Buffer.from(encodedValue, "base64");
+    return JSON.parse(buff.toString("utf-8"));
+  }
+
+  async function handlePostCreation(
+      accountId,
+      blockHeight,
+      blockTimestamp,
+      receiptId,
+      content
+  ) {
+    try {
+      const postData = {
+        account_id: accountId,
+        block_height: blockHeight,
+        block_timestamp: blockTimestamp,
+        content: content,
+        receipt_id: receiptId,
+      };
+
+      // Call GraphQL mutation to insert a new post
+      await context.db.Posts.insert(postData);
+
+      console.log(`Post by ${accountId} has been added to the database`);
+    } catch (e) {
+      console.log(
+          `Failed to store post by ${accountId} to the database (perhaps it is already stored)`
+      );
+    }
+  }
+
+  async function handleCommentCreation(
+      accountId,
+      blockHeight,
+      blockTimestamp,
+      receiptId,
+      commentString
+  ) {
+    try {
+      const comment = JSON.parse(commentString);
+      const postAuthor = comment.item.path.split("/")[0];
+      const postBlockHeight = comment.item.blockHeight;
+
+      // find post to retrieve Id or print a warning that we don't have it
+      try {
+        // Call GraphQL query to fetch posts that match specified criteria
+        const posts = await context.db.Posts.select(
+            { account_id: postAuthor, block_height: postBlockHeight },
+            1
+        );
+        console.log(`posts: ${JSON.stringify(posts)}`);
+        if (posts.length === 0) {
+          return;
+        }
+
+        const post = posts[0];
+
+        try {
+          delete comment["item"];
+          const commentData = {
+            account_id: accountId,
+            receipt_id: receiptId,
+            block_height: blockHeight,
+            block_timestamp: blockTimestamp,
+            content: JSON.stringify(comment),
+            post_id: post.id,
+          };
+          // Call GraphQL mutation to insert a new comment
+          await context.db.Comments.insert(commentData);
+
+          // Update last comment timestamp in Post table
+          const currentTimestamp = Date.now();
+          await context.db.Posts.update(
+              { id: post.id },
+              { last_comment_timestamp: currentTimestamp }
+          );
+          console.log(`Comment by ${accountId} has been added to the database`);
+        } catch (e) {
+          console.log(
+              `Failed to store comment to the post ${postAuthor}/${postBlockHeight} by ${accountId} perhaps it has already been stored. Error ${e}`
```

Comment:
should the failures be logged with `console.error`?

Reply:
@charleslavon This is the pattern used so while working the editor all of the errors show up in the console in debug mode, but maybe we should update all of the handlers to use console.error?